utils/config.go

```diff
@@ -43,6 +43,11 @@ type SSOSetting struct {
 	UserApiEndpoint string
 }
 
+type SSOSettings struct {
+	AllowEmailSignUp bool
```

Comment:
I think it fits here better than anywhere else, because if it you turn off email sign up (and don't have an SSO turned on) you can't login or signup at all.